src/longhorn_rpc_protocol.c

```diff
@@ -11,7 +11,7 @@ int read_full(int fd, void *buf, int len) {
 
         while (readed < len) {
                 ret = read(fd, buf + readed, len - readed);
-                if (ret < 0) {
+                if (ret <= 0) {
                         return ret;
```

Comment:
Not sure if `return 0` is safe enough.
Is the case possible?
```
len = 8192
1st read = 4096
2st read = 0
3st read = 4096 
```

Reply:
No, if the socket connection is open, `read` won't return 0 if the next `read` could read something and return a value greater than zero.

This is uses blocking I/O so `read` will wait until there is something to read.    So the return value of zero effectively means end of file (EOF).   If you were using `read` on a file descriptor that corresponded to a file on disk, the return value of zero will mean you reached the end of that file and every subsequent `read` would also return zero.

The only scenario where `read` could return without reading something on a blocking file descriptor would be if it returned -1 and set `errno` to `EINTR` to indicate the system call was interrupted.  I handle this now. 

I also return `readed` when `ret` equal 0 now and I renamed `readed` to `nread`, both on @yasker 's recommendation.

Reply:
> I also return readed when ret equal 0 now and I renamed readed to nread, both on @yasker 's recommendation.

Yep, returning `readed` makes sense now.